src/pkg/runner/runner.go

```diff
@@ -168,14 +186,49 @@ func (r *Runner) getTask(taskName string) (types.Task, error) {
 	return types.Task{}, fmt.Errorf("task name %s not found", taskName)
 }
 
+func mergeEnv(env1, env2 []string) []string {
+	for _, s1 := range env1 {
+		replaced := false
+		for j, s2 := range env2 {
+			if strings.Split(s1, "=")[0] == strings.Split(s2, "=")[0] {
+				env2[j] = s1
+				replaced = true
+			}
+		}
+		if !replaced {
+			env2 = append(env2, s1)
+		}
+	}
+	return env2
+}
+
+func formatEnvVar(name, value string) string {
+	// replace all non-alphanumeric characters with underscores
+	name = regexp.MustCompile(`[^a-zA-Z0-9]+`).ReplaceAllString(name, "_")
+	name = strings.ToUpper(name)
+	// prefix with INPUT_ (same as GitHub Actions)
+	return fmt.Sprintf("INPUT_%s=%s", name, value)
+}
+
 func (r *Runner) executeTask(task types.Task) error {
 	if len(task.Files) > 0 {
 		if err := r.placeFiles(task.Files); err != nil {
 			return err
 		}
 	}
 
+	defaultEnv := []string{}
+	for name, ip := range task.Inputs {
+		d := ip.Default
+		if d == "" {
+			continue
+		}
+		defaultEnv = append(defaultEnv, formatEnvVar(name, d))
+	}
+
 	for _, action := range task.Actions {
+		action.Env = mergeEnv(action.Env, defaultEnv)
```

Comment:
No need to do this if `defaultEnv` is empty

Reply:
Is it a noop? I read this as `mergeEnv` will loop through `action.Env` copy it into `defaultEnv` and then return it
```
func mergeEnv(env1, env2 []string) []string {
	for _, s1 := range env1 {
		replaced := false
		for j, s2 := range env2 {
			if strings.Split(s1, "=")[0] == strings.Split(s2, "=")[0] {
				env2[j] = s1
				replaced = true
			}
		}
		if !replaced {
			env2 = append(env2, s1)
		}
	}
	return env2
}
```

Reply:
true, noop was not the correct verbiage.

Yes, if `defaultEnv` is empty then this action boils down to `action.Env = action.Env` (after looping through `defaultEnv`, which is `nil` and creating a new slice in memory containing `action.Env`).

The speed cost of this operation is mainly negligible. So imo there is no need to add an additional logic statement to constrict the conditions it runs under.

```
if len(defaultEnv) > 0 {
 action.Env = mergeEnv(action.Env, defaultEnv)
}
```

does not actually perform any (logic wise) operations differently than straight calling:

```
action.Env = mergeEnv(action.Env, defaultEnv)
```

Reply:
> The speed cost of this operation is mainly negligible

I'm slightly trolling bc of @Racer159 's issue https://github.com/defenseunicorns/uds-cli/issues/344 😆. No need to change though, I also believe it's negligible